worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageDriverImpl.java

```diff
@@ -190,4 +249,116 @@ public void loadMappingJob() {
             coverageProcessor.queueMapping(mapping, false);
         }
     }
+
+    /**
+     * This is the most important part of the class. It retrieves the next search in the table
+     * assuming that another thread or application is not currently pulling anything from the table.
+     * If there are no jobs to pull or the table is locked, it returns null
+     *
+     * @return the next search or else null if there are none or if the table is locked
+     */
+    public Optional<CoverageSearch> getNextSearch() {
+        Lock lock = coverageLockWrapper.getCoverageLock();
+        if (lock.tryLock()) {
```

Comment:
Style preference.  I tend not to like
if (something) {
tons of code;
} else {
one line of code directly returning
}

Instead, 
```
if (!lock.tryLock()) {
return Optional.empty();
}
```
Then, the remainder of the code has one less level of indentation

One might even try wrapping the lock with a Resource so that the code can be try with resources, eliminating the explicit finally.